src/cmap/conn/mod.rs

```diff
@@ -73,19 +75,21 @@ impl Connection {
         id: u32,
         address: StreamAddress,
         generation: u32,
-        connect_timeout: Option<Duration>,
-        tls_options: Option<TlsOptions>,
-        handler: Option<Arc<dyn CmapEventHandler>>,
+        mut options: Option<ConnectionOptions>,
     ) -> Result<Self> {
         let conn = Self {
             id,
             generation,
             pool: None,
-            stream_description: None,
             ready_and_available_time: None,
-            stream: Stream::connect(address.clone(), connect_timeout, tls_options)?,
+            stream: Stream::connect(
+                address.clone(),
+                options.as_mut().and_then(|options| options.connect_timeout.take()),
+                options.as_mut().and_then(|options| options.tls_options.take())
```

Comment:
`StreamOptions` could implement `From<ConnectionOptions>` instead of this manual extraction.